### PR TITLE
Openemr fix 5665 vitals being squashed

### LINKED ISSUE #5665: Update Vitals Form To Prevent Inputs From Being Squashed
Currently the Vitals Form is a table that struggles to be responsive when there are many historical records.  We had to add additional fields to comply with ONC requirements (reason codes, interpretation, etc).  This is causing the fields to scrunch up for some browser environments and with large amounts of vitals.  To handle more mobile responsive areas, we are going to show 2 columns of historical data on medium to large devices and hide those columns on smaller device sizes.  

We will display a full table list of the historical data below the save functionality of the form.

### PR DESCRIPTION
**Pull request by adunsulag**
Fixes #5665 
Prevent the form from being squashed up and move the historical data into its own section further down in the form.  Fix some minor twig issues.

Big overhaul of vitals from changing it from smarty to twig format. Cleaned up a lot of the rendering as well so we can both easily extend the vitals form and handle additional fields.  This is prep work to make
the form responsive.

Added some additional extension to twig so we could handle the weird date formats for the vital forms

below added by bradymiller:
Up For Grabs demo (includes the standard demo data) for this PR is at: https://www.open-emr.org/wiki/index.php/Development_Demo#Epsilon_-_Up_For_Grabs_Demo

### CODE REVIEW

**Comment by adunsulag** [comment]
@stephenwaite I'd love your feedback on this if you want to give it a whirl and see if you think it addresses the problems that Nerik was having.

**Review by stephenwaite — APPROVED** [review]

**Comment by sjpadgett** [comment]
I wonder how important the historical table view is to clinicians? From my experience with my doctors they seem to always bring up a graph.
Still though, good idea to put a link to history. clever.

**Comment by bradymiller** [comment]
Up For Grabs demo (includes the standard demo data) for this PR is at: https://www.open-emr.org/wiki/index.php/Development_Demo#Epsilon_-_Up_For_Grabs_Demo

**Comment by bradymiller** [comment]
@sjpadgett , getting to historical view is very helpful and dependent on specialist. For myself, I usually just care about weight changes over time, which is one of the first things i look at when prepping.

**Comment by bradymiller** [comment]
On testing on the demo.

Note the timestamps are messed up for the demo patients and old demo data; this is not a new issue, so can disregard that.

The graphing appears to not be working, for example, should see the bp data graphed here:
image

(can see graphing when click on 'Click here to view and graph all vitals.' in patient summary vitals widget; if have more than 1 data point fro BP then it will show bp; and can click on other rows to see other data graphed )

**Comment by adunsulag** [comment]
@bradymiller Thanks for catching that, I'll work on a fix for the graphing.

**Comment by bradymiller** [comment]
@adunsulag , disregard my report of the broken vitals table. It is working the same as prior to your changes. The reason it was not working in my testing is that graph goes blank if vitals are too close together (ie. less than 1 minute), which is the same behavior as prior to this PR (ie. a preexisting bug). I'll do a more detailed code review of this later today will goal of getting this into patch 2 to give it a bit more time for testing (planning to release patch 1 today).

File: interface/forms/vitals/C_FormVitals.class.php [hunk 1]
```diff
@@ -21,8 +21,10 @@
 use OpenEMR\Common\Uuid\UuidRegistry;
 use OpenEMR\Services\VitalsService;
 use OpenEMR\Services\ListService;
+use OpenEMR\Common\Twig\TwigContainer;
+use Twig\Environment;
```
Comment (bradymiller) [inline interface/forms/vitals/C_FormVitals.class.php:25]: can't see where class in above line is used

File: interface/forms/vitals/C_FormVitals.class.php [hunk 2]
```diff
@@ -21,8 +21,10 @@
...
 
-class C_FormVitals extends Controller
+class C_FormVitals
```
Comment (bradymiller) [inline interface/forms/vitals/C_FormVitals.class.php:27]: 2DV

File: interface/forms/vitals/templates/vitals/vitals.html.twig [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+* @link      http://www.open-emr.org
+* @author    Brady Miller <brady.g.miller@gmail.com>
+* @copyright Copyright (c) 2018 Brady Miller <brady.g.miller@gmail.com>
+* @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+*#}
+<html>
+<head>
+    <title>{{ "Vitals"|xlt }}</title>
+    {{ setupHeader(["datetime-picker","reason-code-widget"]) }}
+<!--Update the version (v flag) to cache bust if you modify this file -->
```
Comment (bradymiller) [inline interface/forms/vitals/templates/vitals/vitals.html.twig:14]: can drop above comment

File: interface/forms/vitals/templates/vitals/vitals.html.twig [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+        ,'validateFailed': window.xl("Please correct the value(s) before proceeding!")
+        ,'invalidField': window.xl("The following field has an invalid value")
+    };
+    if (window.vitalsForm) {
+        window.vitalsForm.init({{ FORM_ACTION|js_url }}, vitalsTranslations);
+    } else {
+        console.error("Failed to find vitalsForm object to initialize javascript vitals form");
+    }
+</script>
+<!--Update the version (v flag) to cache bust if you modify this file -->
```
Comment (bradymiller) [inline interface/forms/vitals/templates/vitals/vitals.html.twig:33]: can drop above comment

File: interface/forms/vitals/templates/vitals/vitals.html.twig [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+                </a>
+            </h2>
+        </div>
+        {% if hasMoreVitals %}
+        <div class="col-8 text-right">
+                <h6 class="h6-responsive"><a href="#patient-vitals-history" class="text-decoration-none">{{ "See More Vital Records History"|xlt }}</a></h6>
+        </div>
+        {% endif %}
+        <div class="col-sm-12">
+            <form id="vitalsForm" name="vitals" method="post" action="{{ FORM_ACTION }}/interface/forms/vitals/save.php">
```
Comment (bradymiller) [inline interface/forms/vitals/templates/vitals/vitals.html.twig:55]: attr

File: interface/forms/vitals/templates/vitals/vitals_growthchart_actions.html.twig [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+{% else %}
+<tr>
+{% endif %}
+    {% if is_edit %}
+    <td colspan='5' class="text-center pt-2">
+    {% else %}
+    <td colspan='2' class="text-center pt-2">
+    {% endif %}
+        <input type="button" class="btn btn-primary" id="pdfchart" value='{{ "Growth-Chart"|xla }} ({{ "PDF"|xlt }})'>
+        <input type="button" class="btn btn-primary" id="htmlchart" value='{{ "Growth-Chart"|xla }} ({{ "HTML"|xlt }})'>
```
Comment (bradymiller) [inline interface/forms/vitals/templates/vitals/vitals_growthchart_actions.html.twig:12]: xla for the PDF and HTML above

File: interface/forms/vitals/templates/vitals/vitals_interpretation_selector.html.twig [hunk 7]
```diff
@@ -0,0 +1,10 @@
+<select name="interpretation[{{ input|attr }}]" class="form-control" id='interpretation_{{ input|attr }}'>
+    <option value=""> </option>
+    {% for option in interpretation_options %}
+        {% if vitalDetails is not null and option.id == vitalDetails.get_interpretation_option_id() %}
+                <option selected="selected" value="{{ option.id|attr }}">{{ option.title|xlt }}</option>
+        {% else %}
+            <option {% if option.is_default %}selected="selected"{% endif%} value="{{ option.id|attr }}">{{ option.title }}</option>
```
Comment (bradymiller) [inline interface/forms/vitals/templates/vitals/vitals_interpretation_selector.html.twig:7]: xlt option.title

**Comment by bradymiller** [comment]
review completed :)

**Pull request merged**